### PR TITLE
Initial Ark v0.10.0 refactor

### PR DESCRIPTION
**Pull request by edevenport**
* Updated dependencies.
* Updated examples.
* Updated the README for Ark v0.10.0 compatibility.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,22 +46,28 @@ This quickstart will describe the installation and configuration of the DigitalO
         --from-literal digitalocean_token=<DIGITALOCEAN_TOKEN>
     ```
 
-4. Update the `examples/10-ark-config.yaml` with the Spaces API URL, bucket, and region and apply the configuration:
+4. Update the `examples/05-ark-backupstoragelocation.yaml` with the DigitalOcean Spaces API URL, bucket, and region and apply the `BackupStorageLocation` configuration.
```
Comment (nrb) [inline README.md]: The location of these files will be changing with PR https://github.com/heptio/ark/pull/1045, which is intended to go out with v0.10.0. This should mostly be a docs change, though.

Comment (nrb) [reply README.md]: Ah, I was skimming this in order and missed that this plugin ships with it's own YAML - disregard!

File: examples/10-deployment.yaml [hunk 2]
```diff
@@ -19,11 +19,12 @@ spec:
       serviceAccountName: ark
       containers:
         - name: ark
-          image: gcr.io/heptio-images/ark:latest
+          image: gcr.io/heptio-images/ark:v0.10.0-beta.1
```
Comment (nrb) [inline examples/10-deployment.yaml]: Are you planning to update this with the final version before merging?

**Comment by nrb** [comment]
Looks good to me! Thanks for getting this updated. We cut a `0.9.x` branch for the examples so that people can keep using them with older releases, but your branching/image tagging strategy is up to your team.

File: README.md (hunk 1 above)
Comment (edevenport) [reply README.md]: I do link to the Ark pre-requisite docs, but also include the pre-req YAML with the plugin for ease of deployment. A user won't need to leave the plugin doc to get a functional environment.

File: examples/10-deployment.yaml (hunk 2 above)
Comment (edevenport) [reply examples/10-deployment.yaml]: Yes, the plan is to update before I merge the PR. I just needed this for testing.

**Comment by edevenport** [comment]
@nrb Thank you so much for the review. Cutting a `v0.9.x` branch is a good idea - I'll follow that approach. Thanks again.

**Pull request merged**